### DISCUSSION TITLE
How to implement beam search in JAX/Flax?

### DISCUSSION THREAD

**Discussion by marcvanzee** [google/flax#917] [Q&A] [answered]

**Comment by marcvanzee** [comment, marked as answer]
We currently have implemented this in our WMT example in [decode.py](https://github.com/google/flax/blob/master/examples/wmt/decode.py). It presumes a function that takes integer tokens and returns logits. It has been optimized for TPUs.
